qa/rpc-tests/test_framework/test_framework.py

```diff
@@ -86,13 +86,9 @@ def join_network(self, id = 1):
         # Join the (previously split) adjanced nodes in linear network together: 0-1 >-< 2-3
         assert self.is_network_split
         connect_nodes_bi(self.nodes, id, id + 1)
+        sync_blocks([self.nodes[id], self.nodes[id + 1]])
```

Comment:
Given that most of the times we rely on a full sync, can we give the possibility to choose how to perform the join? For instance:
- input parameter for `join_network()` to sync blocks only, mempool only, sync all
- by default, sync_all

Reply:
I'll just add my point of view here, but definitely not a blocking thing for me.
I like @ptagl's suggestion, it adds more flexibility for test developers and it allows writing more uniform and clean tests, with minimum impact on the code.

Reply:
I modified the function to improve its flexibility. Now synchronization of blocks and mempools are optional, with the former enabled by default, and the latter disabled.